processing/src/main/java/org/apache/druid/query/filter/InDimFilter.java

```diff
@@ -80,7 +80,8 @@
 {
   // determined through benchmark that binary search on long[] is faster than HashSet until ~16 elements
   // Hashing threshold is not applied to String for now, String still uses ImmutableSortedSet
-  public static final int NUMERIC_HASHING_THRESHOLD = 16;
+  public static final int NUMERIC_HASHING_THRESHOLD =
+      Integer.parseInt(System.getProperty("druid.query.filter.inDimFilter.numericHashingThreshold", "1"));
```

Comment:
I don't think we should have a property for this, but if we did, it should be set through Guice injection rather than through System.getProperty. That way, it'd work in either a runtime props file or a system property.

We won't actually see "1" in practice AFAIK, since `optimize()` will get run, and that'll turn the In filter into a Selector filter.

So if this is the right default, I think it'd be better to remove the code.

Reply:
ha I didn't look at the optimize code in the `InDimFilter` before.
`SelectorDimFilter.optimize()` produces an `InDimFilter` and `InDimFilter.optimize()` produces a `SelectorDimFilter`. One of those must be wrong 😅 

Either way, it probably doesn't make a difference with 1 element.

I put the default in mainly because of my paranoia, just in case this causes a perf degradation for a specific shape of query that isn't covered by my benchmarks. I'll run them a few more times and then just drop this code path.

Reply:
> One of those must be wrong 😅

SelectorDimFilter calls `new InDimFilter(dimension, Collections.singleton(value), extractionFn, filterTuning).optimize()`, so you get another SelectorDimFilter back 🙂

There's no comment about why, but I assume it's because InDimFilter has the `optimizeLookup()` code.

> I put the default in mainly because of my paranoia, just in case this causes a perf degradation for a specific shape of query that isn't covered by my benchmarks.

People generally aren't going to have the patience to research each of these settings, so it's usually better if we do some diligence to find something that should work (relatively) universally. If that involves removing code paths then it also helps us reduce the amount of code that needs to be tested.